evalml/automl/engine/sequential_engine.py

```diff
@@ -0,0 +1,28 @@
+from copy import copy
+
+from evalml.automl.engine import EngineBase
+
+
+class SequentialEngine(EngineBase):
+    """The default engine for the AutoML search. Trains and scores pipelines locally, one after another."""
+
+    def evaluate_batch(self, pipelines):
+        """Evaluate a batch of pipelines using the current dataset and AutoML state.
+
+        Arguments:
+            pipelines (list(PipelineBase)): A batch of pipelines to be fitted and evaluated.
+
+        Returns:
+            list (int): a list of the new pipeline IDs which were created by the AutoML search.
+        """
+        if self.X_train is None or self.y_train is None:
+            raise ValueError("Dataset has not been loaded into the engine.")
+        new_pipeline_ids = []
+        pipelines = copy(pipelines)
```

Comment:
You are right! We clone the pipeline in `train_and_score_pipeline`, so that we can train one on each CV fold.

The reason I am copying this list is that `pipelines.pop(0)` modifies the list. Because python passes by reference this means its the list from up the stack which is modified. That's not great behavior IMO, too confusing.

So the question is, how can we have this code iterate through a list of pipelines without altering the parent pipeline list from up the stack?

Your comment is making me think we should use an index instead of copying/popping. I'll give that a shot.